Add env var to export a hint file to optimize ASAR ordering

bengotow opened the pull request:
Overview here: https://github.com/atom/atom/issues/10163

zeke commented on docs/api/environment-variables.md:
```diff
@@ -25,6 +25,12 @@ Starts the process as a normal Node.js process.
 
+## `ELECTRON_LOG_ASAR_READS`
+
+When Electron reads from an ASAR file, log the the read offset and file path
```
typo: the the

zeke commented on docs/api/environment-variables.md:
```diff
@@ -25,6 +25,12 @@ Starts the process as a normal Node.js process.
+## `ELECTRON_LOG_ASAR_READS`
+
+When Electron reads from an ASAR file, log the the read offset and file path
+to the system `tmpdir`. Th resulting file can be provided to the ASAR module
```
typo: Th

etiktin commented:
Some asar resources are loaded from the c++ side ([example](https://github.com/atom/electron/blob/c1673eb666540b2891794bcb2f477f4b0cee78de/atom/common/api/atom_api_native_image.cc#L137)). Will this code log them as well?

bengotow commented:
Hey folks thanks for the feedback—sorry I dropped the ball on this one. I'll address those typos! @etiktin, this isn't currently logging resources loaded from the C++ side. I'm not sure if it's possible to log those to the same file, and ensure that they're written to the logged in the correct order...

zcbenz commented:
@bengotow Do you mind rebasing this on master branch? Recently we have converted all our CoffeeScript files to JavaScript.

bengotow commented:
Glad to see everything is going to ES6. I've converted my changes to JS and updated the pull request. Also double-checked to make sure it still works correctly.

zcbenz commented:
:+1:
